tools/clang/include/clang/SPIRV/SpirvBuilder.h

```diff
@@ -963,17 +970,47 @@ SpirvBuilder::setDebugSource(uint32_t major, uint32_t minor,
 SpirvInstruction *
 SpirvBuilder::addExecutionMode(SpirvFunction *entryPoint, spv::ExecutionMode em,
                                llvm::ArrayRef<uint32_t> params,
-                               SourceLocation loc, bool useIdParams) {
+                               SourceLocation loc) {
   SpirvExecutionMode *mode = nullptr;
-  SpirvExecutionMode *existingInstruction =
+  SpirvExecutionModeBase *existingInstruction =
+      mod->findExecutionMode(entryPoint, em);
```

Comment:
Find doesn't check the params are the same. Should this check all the instruction operands are the same before returning the existing one?

Reply:
I'm keeping the existing behaviour. You are probably right. I'll open up a follow up issue for that. I see that as outside the scope of this PR, which is about refactoring existing behaviour.